week4/week4Pllayground.playground/Contents.swift

```diff
@@ -0,0 +1,241 @@
+func evenNumbersArray (arrayOfInt:[Int]) -> [Int] {
+  var evenNumbers: [Int] = []
+  for int in arrayOfInt {
+    if int % 2 == 0 {
+      evenNumbers.append(int)
+    }
+  }
+  return evenNumbers
+}
+
+var evenNums = evenNumbersArray(arrayOfInt: nums)
+print(evenNums)
+
+// Create a function called sumOfArray that takes a parameter of [Int?] and returns an Int. The function should return the sum of the array values passed that are not nil. Call the function passing the numsWithNil array, and print out the results.
+
+func sumOfArray (num: [Int?]) -> Int {
+  var sum = 0
+  for num in num {
+    if let unwrappedNum = num {
+      sum += unwrappedNum
+    }
+  }
+  return sum
+  }
+
+sumOfArray(num: numsWithNil)
+
+////- Create a function called commonElementsSet that takes two parameters of [Int] and returns a Set<Int> (set of Int.) The function will return a Set<Int> of the values in both arrays.
+//Call the function commonElementsSet passing the arrays numsBy2, numsBy4, and print out the results.
+
+func commonElementsSet(array1: [Int], array2: [Int]) -> Set<Int> {
+  let array1 = Set(array1)
+  let array2 = Set(array2)
+  
+  let commonSet = array1.intersection(array2)
+  
+  return commonSet
+}
+
+commonElementsSet(array1: numsBy2, array2: numsBy4)
+
+//d) Create a struct called Square that has a stored property called sideLength and a computed property called area. Create an instance of Square and print out the area.
+
+struct Square {
+  var sideLength: Double
+  var area: Double {
+    return sideLength * sideLength
+  }
+  
+  init(sideLength: Double) {
+    self.sideLength = sideLength
+  }
+
+}
+
+var square = Square(sideLength: 48)
+
+//above and beyond
+//Note: you don’t need to complete Above and Beyond sections in order to pass the homework. If you need to, you can stop here. But if you have the time and bandwidth, the A&B sections will let you challenge yourself further.
+//Create a protocol called Shape with a calculateArea() -> Double method. Create two structs called Circle and Rectangle that conform to the protocol Shape. Both Circle and Rectangle should have appropriate stored properties for calculating the area.
+//Create instances of Circle and Rectangle and print out the area for each.
+
+
+protocol Shape {
+  func calculateArea() -> Double
+}
+
+struct Circle: Shape {
+  var radius: Double = 0.0
+
+  func calculateArea() -> Double {
+    var area:Double {
+       (.pi * radius * radius)
+    }
+    return area
+  }
+  init(radius: Double) {
+    self.radius = radius
+  }
+}
+
+struct Rectangle: Shape {
+  var length: Double
+  var width: Double
+  
+  func calculateArea() -> Double {
+    var area: Double {
+      length * width
+    }
+    return area
+    }
+  init(length: Double, width: Double) {
+    self.length = length
+    self.width = width
+  }
+  }
+
+extension Shape {
+  func calculateVolume() -> Double {
+    return 0
+  }
+}
+
+struct Sphere{
```

Comment:
`Sphere` is not confirming to the `Shape` protocol. It implements its methods but it doesn't conform to the protocol.